guides/plugins/plugins/storefront/add-dynamic-content-via-ajax-calls.md

```diff
@@ -0,0 +1,138 @@
+# Add dynamic content via ajax calls
+
+This guide will show you how to add dynamic content to your storefront.
+It combines and builds upon the the guides about [adding custom Javascript](./add-custom-javascript.md) and [adding a custom controller](./add-custom-controller.md), so you should probably read them first.
+
+## Setting up the Controller
+
+For this guide we will use a very simple controller that returns a timestamp wrapped in the JSON format.
+
+As mentioned before this guide builds up upon the [adding a custom controller](./add-custom-controller.md) guide.
+This means that this article will only cover the differences between returning a template and a `JSON` response and making it accessible to `XmlHttpRequests`.
+
+```php
+<?php declare(strict_types=1);
+
+namespace SwagBasicExample\Storefront\Controller;
+
+use Shopware\Core\Framework\Routing\Annotation\RouteScope;
+use Shopware\Storefront\Controller\StorefrontController;
+use Symfony\Component\HttpFoundation\JsonResponse;
+use Symfony\Component\Routing\Annotation\Route;
+
+/**
+ * @RouteScope(scopes={"storefront"})
+ */
+class ExampleController extends StorefrontController
+{
+   /**
+    * @Route("/example", name="frontend.example.example", defaults={"XmlHttpRequest"=true}, methods={"GET"})
+    */
+    public function showExample(): JsonResponse
+    {
+        return new JsonResponse(['timestamp' => gmdate("Y-m-d\TH:i:s\Z")]);
```

Comment:
According to php docs the timestamp parameter in `gmdate` is nullable since php 8.0. As some fellows are still on 7.2 or 7.4 regarding their projects, I suggest to change the snippet to `date(\DateTimeInterface::W3C)`

Reply:
```suggestion
        return new JsonResponse(['timestamp' => date(\DateTimeInterface::W3C)]);
```
So like this? I'm mainly a frontend developer and don't know much about php

Reply:
Yes :)

Reply:
Just a little suggestion ;) 

```suggestion
        return new JsonResponse(['timestamp' => \date(\DateTimeInterface::W3C)]);
```

Reply:
Good one :D

Reply:
Wasn't it you, who created a huge pr for straightening these global references in shopware\platform ? :laughing: 
Since then, I'm having an eye for these :rofl:

Reply:
I was … and I have been replaced by an automated tool :/ @mitelg 

Great to hear that I influenced at least one of you 😁

Reply:
best would be using the `DateTime` object!

Reply:
make a suggestion and i'll use that

Reply:
did that :blush:

Reply:
Ok i'll just trust you that it works

Reply:
yeah sure :D the object is better